stl/src/atomic_wait.cpp

```diff
@@ -190,6 +190,24 @@ namespace {
         }
     }
 #endif // _ATOMIC_WAIT_ON_ADDRESS_STATICALLY_AVAILABLE
+
+    _NODISCARD unsigned char __std_atomic_compare_exchange_128_fallback(_Inout_bytecount_(16) long long* _Destination,
+        _In_ long long _ExchangeHigh, _In_ long long _ExchangeLow,
+        _Inout_bytecount_(16) long long* _ComparandResult) noexcept {
+        static SRWLOCK _Mtx = SRWLOCK_INIT;
+        _SrwLock_guard _Guard{_Mtx};
+        if (_Destination[0] == _ComparandResult[0] && _Destination[1] == _ComparandResult[1]) {
+            _ComparandResult[0] = _Destination[0];
+            _ComparandResult[1] = _Destination[1];
+            _Destination[0]     = _ExchangeLow;
+            _Destination[1]     = _ExchangeHigh;
+            return true;

```

Comment:
It's somewhat strange to convert `true` and `false` to `unsigned char`; should the return type be `bool` or should the return values be `1` and `0`?